pkg/service/endpoints_controller.go

```diff
@@ -150,18 +150,35 @@ func endpointsEqual(eps *api.Endpoints, endpoints []api.Endpoint) bool {
 	return true
 }
 
-// findPort locates the container port for the given manifest and portName.
-func findPort(pod *api.Pod, portName util.IntOrString) (int, error) {
-	firstContainerPort := 0
-	if len(pod.Spec.Containers) > 0 && len(pod.Spec.Containers[0].Ports) > 0 {
-		firstContainerPort = pod.Spec.Containers[0].Ports[0].ContainerPort
+func findDefaultPort(pod *api.Pod, servicePort int) (int, bool) {
+	foundPort := 0
+	portMap := map[string]int{}
+	for _, container := range pod.Spec.Containers {
+		for _, port := range container.Ports {
+			if port.Name == "" {
+				foundPort = port.ContainerPort
```

Comment:
Wouldn't you want to break here, but you probably want a double break (break-to-label) ?